### PR TITLE
[Bugfix] Per-key accumulation strategy for Qwen3-TTS codec frame outputs

### PR DESCRIPTION
**Pull request by linyueqian**
## Summary

`OutputProcessor._consolidate_multimodal_tensors` picks one `TensorAccumulationStrategy` from the request's output modality (`get_accumulation_strategy`) and applies it uniformly to every accumulated tensor key of that modality (`MultimodalPayload.consolidate_tensors` → `_consolidate_tensor_list`). Qwen3-TTS's talker stage emits two codec-frame keys under the AUDIO modality that are not waveform chunks:

- `codes.audio` is a `[rows, num_codebooks]` matrix that grows one row per decode step along dim 0 (the prefill contributes a larger initial block, each decode step one more row).
- `codes.ref` is the constant reference-context matrix the talker re-emits unchanged at every step.

Forced through AUDIO's waveform-tuned default (`CONCAT_LAST`, i.e. `torch.cat(dim=-1)`), both are corrupted whenever a stage carrying them consolidates its output under the AUDIO modality:

1. `codes.audio`'s dim-0 growth across steps makes the dim=-1 concat raise (`RuntimeError`). The `except` branch only special-cased the literal key `"audio"`, so `"codes.audio"` fell into the warn-and-keep-last-tensor path — the consolidated payload became only the final one-row step instead of the full sequence.
2. `codes.ref`'s per-step blocks share the same last-dim size, so the dim=-1 concat does *not* raise — it silently concatenates N identical copies of the reference matrix side by side instead of keeping one copy.

Both are demonstrated with synthetic tensors in the new test (no GPU/model needed).

## Fix

Let the accumulation strategy be resolved per `(modality, key)` instead of per-modality-only:

- `get_accumulation_strategy` now takes an optional `key` argument and checks a new per-key override registry (`register_key_accumulation_strategy`) before falling back to the modality-wide default. This avoids a hardcoded key list in the shared consolidator — a pipeline module registers the override for the keys it knows about.
- Qwen3-TTS's `pipeline.py` registers `"codes.audio" -> CONCAT_DIM0` and `"codes.ref" -> REPLACE` at import time, so any request whose accumulated output is consolidated under the AUDIO modality gets the correct per-key behavior for these two keys while everything else (real waveform chunks) keeps the existing `CONCAT_LAST` default.
- `MultimodalPayload.consolidate_tensors` now takes the modality directly and resolves a strategy per key via `get_accumulation_strategy(modality, key)`, rather than being handed one strategy for the whole payload.
- Generalized the concat-failure fallback in `_consolidate_tensor_list`: a `CONCAT_LAST` failure still falls back to flattening each chunk (this is a real need for waveform chunks whose non-last dims can drift between async-chunk steps), but a failure under any *other* strategy now raises with the offending key name instead of silently keeping only the last chunk. Previously this silent fallback applied to every key except the literal string `"audio"`, so any other mismatched key (codec frames included) was corrupted instead of failing loudly.

This also removes a small piece of dead code: the previous fallback retried `torch.cat(tensor_list, dim=-1)` unconditionally before flattening, which is the exact call `_cat_tensors` already just made for `CONCAT_LAST` (a guaranteed no-op retry).

No existing pipeline is affected at runtime today — the two-stage Qwen3-TTS pipeline's talker stage uses `engine_output_type="latent"`, so its codec-frame keys are never consolidated under the AUDIO modality in the shipped topology. This fix is needed for a variant that finishes on an AUDIO-typed stage that forwards `codes.audio`/`codes.ref` directly (see below).

## Test

`tests/engine/test_output_modality.py` (same bypass-package-init loading the file already uses, so no vLLM install is required):

- `test_get_accumulation_strategy_key_override_precedence` — a registered override wins for its key; every other key keeps the modality default.
- `test_consolidate_tensors_default_audio_strategy_corrupts_codec_frame_keys` — reproduces both failure modes above with synthetic tensors and no override registered.
- `test_consolidate_tensors_with_qwen3_tts_key_overrides_is_correct` — the same synthetic payload consolidates correctly once the Qwen3-TTS overrides are registered.
- `test_consolidate_tensors_raises_with_key_name_instead_of_silently_keeping_last` — a concat failure under a non-`CONCAT_LAST` strategy now raises with the key name instead of silently keeping stale data.

Commands run locally:

```
ruff check tests/engine/test_output_modality.py vllm_omni/model_executor/models/qwen3_tts/pipeline.py vllm_omni/outputs/mm_outputs.py vllm_omni/outputs/output_modality.py vllm_omni/outputs/output_processor.py
ruff format --check <same files>
```

Both pass with no changes needed. `mypy` (via `pre-commit run mypy-3.10`) reports the same 8 pre-existing errors on these files before and after this change (`vllm_omni/model_executor/models/` is excluded from the mypy ratchet, so `pipeline.py` isn't checked) — no new errors introduced.

`pytest tests/engine/test_output_modality.py` passes (7/7, including the 4 new tests) against a local build of just the touched `vllm_omni/outputs/*` modules; against the pre-fix modules the same test file fails to even collect (`register_key_accumulation_strategy` does not exist yet), confirming the new tests catch this bug. I don't have a CUDA/vLLM machine available right now to run the full suite against a real install, so I'd appreciate CI running the broader `tests/engine/` and Qwen3-TTS suites here.

Opening as a draft while that runs.

### CODE REVIEW

**Review by Sy0307 — APPROVED** [review]
LGTM.
